packages/swayze/lib/src/widgets/headers/gestures/resize_header/header_edge_mouse_listener.dart

```diff
@@ -0,0 +1,278 @@
+        displacement < 0 &&
+        localPixelOffset < frozenExtent + 2;
+
+    // we need to ignore the `displacement` in frozen headers since they are
+    // fixed when we scroll.
+    if (!ignoreDisplacement) {
+      localPixelOffset += displacement.abs();
+    }
+
+    localPixelOffset = localPixelOffset.floorToDouble();
+
+    final offsets = axisContext.value.headersEdgesOffsets;
+
+    if (offsets.containsKey(localPixelOffset)) {
+      final headerEdgeInfo = offsets[localPixelOffset]!;
+
+      resizeNotifier.value = ResizeHeaderDetails(
+        edgeInfo: headerEdgeInfo,
+        axis: axis,
+      );
+
+      return true;
+    }
+
+    return false;
+  }
+
+  double minExtent(Axis axis) =>
+      axis == Axis.horizontal ? kDefaultCellWidth : kDefaultCellHeight;
+
+  /// Inserts an [OverlayEntry] to the current [OverlayState] with resize line
+  /// on it at the header edge that is being hovered.
+  void _handleOnPointerDown(PointerDownEvent event) {
+    if (!resizeNotifier.isHoveringHeaderEdge) {
+      return;
+    }
+
+    final details = resizeNotifier.value!;
+    final axis = details.axis;
+
+    final initialOffset = _getOffsetPositionForAxis(event.localPosition, axis) +
+        details.edgeInfo.displacement;
+
+    final minOffset =
+        initialOffset - (details.edgeInfo.width - minExtent(axis));
+
+    resizeNotifier.value = details.copyWith(
+      offset: Wrapped.value(initialOffset),
+      initialOffset: Wrapped.value(initialOffset),
+      minOffset: Wrapped.value(minOffset),
+    );
+
+    resizeLineOverlayManager.insertEntries(context);
+  }
+
+  /// Updates the resize line position by adding [event.delta] to
+  /// its current position.
+  void _handleOnPointerMove(PointerMoveEvent event) {
+    if (!resizeNotifier.isResizingHeader) {
+      return;
+    }
+
+    final details = resizeNotifier.value!;
+    final axis = details.axis;
+    final offset = details.offset!;
+
+    final newOffset = offset + _getOffsetPositionForAxis(event.delta, axis);
+
+    resizeNotifier.value = details.copyWith(offset: Wrapped.value(newOffset));
+  }
+
+  /// Sets the header extent of the header that has been resized and removes the
+  /// [OverlayEntry] that contains the resize line.
+  void _handleOnPointerUp(PointerUpEvent event) {
+    if (!resizeNotifier.isResizingHeader) {
+      return;
+    }
+
+    final value = resizeNotifier.value!;
+    final axis = value.axis;
+
+    final headerController = internalScope.controller.tableDataController
+        .getHeaderControllerFor(axis: axis);
+
+    final index = value.edgeInfo.index;
+    final initialOffset = value.initialOffset!;
+    final offset = value.offset!;
+    final extent = value.edgeInfo.width;
+
+    double newExtent;
+
+    if (offset < initialOffset) {
+      newExtent = max(minExtent(axis), extent - (initialOffset - offset));
+    } else {
+      newExtent = extent + (offset - initialOffset);
+    }
+
+    newExtent = newExtent.floorToDouble();
```

Comment:
Why are we using `floorToDouble()`?

Reply:
when saving the header edge offsets after changing an extent, it can lead to values like "624.232132134", this prevents that. You could argue that then it's not as precise as it should be. Well tbh, after testing it, if there is a difference, it's not noticeable. Also, our BE does not work with decimals in sizes

Reply:
Then, would it make sense to use `int`s instead?

Reply:
since we would have to refactor the other variables too (like `offsets`, `sizes`, etc), I think it's something we can do in the future